include/SDL3/SDL_video.h

```diff
@@ -66,11 +66,9 @@ typedef struct
 {
     SDL_DisplayID displayID;    /**< the display this mode is associated with */
     Uint32 format;              /**< pixel format */
-    int pixel_w;                /**< width in pixels (used for creating back buffers) */
-    int pixel_h;                /**< height in pixels (used for creating back buffers) */
-    int screen_w;               /**< width in screen coordinates (used for creating windows) */
-    int screen_h;               /**< height in screen coordinates (used for creating windows) */
-    float display_scale;        /**< scale converting screen coordinates to pixels (e.g. a 2560x1440 screen size mode with 1.5 scale would have 3840x2160 pixels) */
+    int w;                      /**< width */
+    int h;                      /**< height */
+    float pixel_density;        /**< scale converting size to pixels (e.g. a 1920x1080 mode with 2.0 scale would have 3840x2160 pixels) */
```

Comment:
Would it be possible to split this into separate variables for width and height? The RISC OS and 3DS video drivers are both likely to benefit from this.

Reply:
It would add a fair amount of complexity, both internally and for the application, understanding how to handle non-uniform scale. Can you provide an example of how this could happen? Both of those platforms currently use normal width and height and pixel density of 1.0.

Reply:
- On RISC OS, modes have separate X and Y eigen factor values of 0-3 (for DPIs of 180, 90, 45 and 22.5). Square pixel modes are the most common, but some older numbered modes like mode 15 use EX1 EY2.
- On the 3DS, all models except the Old 2DS support a higher resolution 800x240 mode for the top screen that can be used instead of the regular 400x240. It can be enabled using the `gfxSetWide()` function in libctru.

Reply:
In that case, what should be reported for `pixel_density` in those modes?